test/functional/interface_usdt_validation.py

```diff
@@ -123,6 +126,10 @@ def handle_blockconnected(_, data, __):
             assert_equal(0, event.sigops)  # no sigops in coinbase tx
             # only plausibility checks
             assert event.duration > 0
+            # generatetoaddress (mining and connecting) takes longer than
+            # connecting the block. In case the duration unit is off, we'll
+            # detect it with this assert.
+            assert event.duration < generatetoaddress_duration[block_hash]
```

Comment:
`In case the duration unit is off, we'll detect it with this assert`

Looks like this detects a change from a less to a more granular duration unit.  Would this detect a change from more granular to less granular (e.g. from nanoseconds back to microseconds)?  Unlikely to encounter this though, now that the cast to nanoseconds is occurring.